src/simple_experiments/view/patient_list.cljs

```diff
@@ -112,12 +112,22 @@
                :let [last? (= (inc i) (count results))]]
            ^{:key (str (random-uuid))}
            [c/touchable-opacity
-            {:on-press #(do
-                          (dispatch [:set-active-card
-                                     (:uuid @active-card)
-                                     :pending-association])
-                          (dispatch [:set-active-patient-id (:id patient)])
-                          (dispatch [:show-bp-sheet]))}
+            {:on-press #(do (dispatch (let [{:keys [uuid six-digit-id]} @active-card
+                                           status (cond
+                                                    uuid
+                                                    :pending-association
+
+                                                    (simple-card/has-six-digit-id? patient six-digit-id)
+                                                    :associated
+
+                                                    (simple-card/pending-association? @active-card)
+                                                    :pending-association
+
+                                                    :else
+                                                    :pending)]
+                                       [:set-active-card uuid six-digit-id status]))
```

Comment:
Please extract this `cond` into a function :) Also, this doesn't feel exhaustive to me. Let's go over this together.

Reply:
cool